imageio/plugins/spe.py

```diff
@@ -533,37 +381,47 @@ def extract_metadata(meta: Mapping, char_encoding: str = "latin1"):
         meta.pop("exposure_sec")
 
 
-class SpeFormat(Format):
-    """See :mod:`imageio.plugins.spe`"""
+class SpePlugin(PluginV3):
+    def __init__(self, request: Request, check_filesize: bool = True) -> None:
```

Comment:
This is not so easy as I initially thought.

What you suggest would mean to set the default arguments of `metadata()` to `None` in order to be able to detect whether the user changed them and then define the actual defaults via `if arg is None` in the method, hiding them from the user/documentation. I don't think that this is worth just to introduce anyhow deprecated parameters.

What can be done easily, though, is 

1. value stored in `plugin.kwarg` (user input to `plugin.__init__`)
1. user input to `plugin.metadata(...)`
1. default of `plugin.metadata(...)`

This would preserve backward compatibility and explicit default arguments. Nobody should be passing arguments to `__init__()` and `metadata()` in the same code anyhow – either they still use v2 or already v3.

Reply:
I implemented the latter in 0ff1db8e13329ad868fd75eecf309cd5e98d22a2.

Reply:
Theoretically, users of v3 can do something the following (which hopefully never happens, because it is indeed not super useful):

```python
with iio.imopen("fancy.spe", plugin="SPE", char_encoding="utf-8") as file:
    meta = file.metadata(char_encoding="iso8859-1")
```

Nevertheless, with this syntax, I would have expected ImageIO to use iso 8859-1 for the data. However, given that this is a real oddball, I think it will be okay to have the constructor's `char_encoding` take preference if that makes implementation easier.

I will take a look when I do the next round of code review - currently schedules for the end of this week.

Reply:
I see that this is possible, but I would expect

- v2 API users / legacy code not to pass any arguments to `metadata()` since this did not exist in v2
- v3 API users to properly call the new API, omitting any deprecated parameters existing only for v2 compatibility

I'd say, anyone using the code above is doing risky stuff intentionally and has to deal with the fallout themselves ;-)